src/components/v2/V2Create/forms/TokenForm/index.tsx

```diff
@@ -207,6 +222,7 @@ export default function TokenForm({
   )
 
   const onTokenFormSaved = useCallback(() => {
+    const weight = tokenForm.getFieldValue('weight')
```

Comment:
**question** is there a reason we need to store the value in component state _and_ the form state? Could we potentially remove the form state and rely solely on the component state variable?

Reply:
Inputs are whack as states, the easiest way to deal with the lagging states I've found is just using the form. If you know a way to get the typing smooth in the inputs as state-only would love to know

Reply:
does setting the `Input` `value` prop to the state variable, and using the `Input` `onChange` to set the state variable, work?

def more overhead but I think its better than having 2 sources of truth for the data, if it's possible

Reply:
Yep not sure what I was doing but that's working haha. Done